common/etc/nginx/include/s3gateway.js

```diff
@@ -449,16 +451,9 @@ function _escapeURIPath(uri) {
  * @private
  */
 function _isDirectory(path) {
-    if (path === undefined) {
-        return false;
-    }
-    const len = path.length;
-
-    if (len < 1) {
-        return false;
-    }
+    if (!path) return false;
```

Comment:
**Annotation**:
Not a big deal, but the previous implementation didn't account for `null`.  Perhaps that was on purpose? Either way, if we assume `null`, `undefined`, or `""` are all "not a directory" this function gets a lot simpler.